### PR TITLE
Issue #11494: Added LITERAL_SYNCHRONIZED token support to WhitespaceA…

### LINKED ISSUE #11494: WhitespaceAfter should support LITERAL_SYNCHRONIZED token
https://checkstyle.org/config_whitespace.html#WhitespaceAfter

I have downloaded the latest cli from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

**How it works Now:**

```bash
/var/tmp $ javac SynchronizedSpace.java
/var/tmp $

/var/tmp $ cat config.xml
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="WhitespaceAfter">
            <property name="tokens" value="LITERAL_SYNCHRONIZED"/>
        </module>
    </module>
</module>

/var/tmp $ cat SynchronizedSpace.java
public class SynchronizedSpace {
    public static void main(String[] args) {
        synchronized(SynchronizedSpace.class) {
            System.out.println("started");
        }
    }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.1-all.jar -c config.xml SynchronizedSpace.java
com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module TreeWalker - Token "LITERAL_SYNCHRONIZED" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:476)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:201)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:403)
        at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:330)
        at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:189)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:126)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Token "LITERAL_SYNCHRONIZED" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck
        at com.puppycrawl.tools.checkstyle.TreeWalker.registerCheck(TreeWalker.java:223)
        at com.puppycrawl.tools.checkstyle.TreeWalker.setupChild(TreeWalker.java:133)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:201)
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:471)
        ... 5 more
Checkstyle ends with 1 errors.
```

**Is your feature request related to a problem? Please describe.**
It would be nice if checkstyle could detect missing space after `synchronized` keyword.

**Describe the solution you'd like**
WhitespaceAfter could support new literal type - `LITERAL_SYNCHRONIZED`

**Additional context**
Google Java Style Guide requires whitespace separation of `(` from any reserved words
See https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace
>1. Separating any reserved word, such as `if`, `for` or `catch`, from an open parenthesis (`(`) that follows it on that line

#8122 seems related. This feature request is part of it.

#11489 and #11493 are similar requests for `try`/`finally` keywords support

### PR DESCRIPTION
**Pull request by Kevin222004**
resolve #11494
Diff Regression config: https://gist.githubusercontent.com/Kevin222004/174f62d3ea5815382dc7aaaa2620238f/raw/ce9b9bc1e9383da9d731702b8e666905998021ee/config.xml
Diff Regression patch config: https://gist.githubusercontent.com/Kevin222004/5ee7cf7d80854a2aa365bcfc016339c2/raw/516423eb9c7e5bc369c70e74a6b9c26ed50bf582/patch_config.xml

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java [hunk 1]
```diff
@@ -100,7 +102,12 @@
  *
  *      try {} catch (Error e){} finally {} // OK
  *      try {} catch (Error e){} finally{} // violation ''finally' is not followed by whitespace'
- *  }
+ *
+ *  public void check3() {
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java:106]: Remove method usage

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java (hunk 1 above)
Comment (Kevin222004) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: done

File: src/xdocs/config_whitespace.xml [hunk 2]
```diff
@@ -2986,7 +2990,11 @@ public void myTest() {
 
     try {} catch (Error e){} finally {} // OK
     try {} catch (Error e){} finally{} // violation ''finally' is not followed by whitespace'
+
+    synchronized(this) { } // violation ''synchronized' is not followed by whitespace'
+    synchronized (this) { }
 }
+
           </pre>
```
Comment (romani) [inline src/xdocs/config_whitespace.xml:2998]: Remove extra change

**Review by romani — CHANGES_REQUESTED** [review]

**Comment by romani** [comment]
please generate check regression report. You will need to use a patch config, see instructions at https://github.com/checkstyle/contribution/tree/master/checkstyle-tester#report-generation

You can see an example at https://github.com/checkstyle/checkstyle/pull/11491

File: src/xdocs/config_whitespace.xml (hunk 2 above)
Comment (Kevin222004) [reply src/xdocs/config_whitespace.xml:2997]: done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java [hunk 3]
```diff
@@ -100,7 +102,11 @@
  *
  *      try {} catch (Error e){} finally {} // OK
  *      try {} catch (Error e){} finally{} // violation ''finally' is not followed by whitespace'
+ *
+ *      synchronized(this) { } // violation ''synchronized' is not followed by whitespace'
+ *      synchronized (this) { }
  *  }
+ *
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java:109]: Remove unnecessary change

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java (hunk 3 above)
Comment (Kevin222004) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: done

**Comment by romani** [comment]
Looks good,  please generate report.

**Comment by Kevin222004** [comment]
GitHub, generate report

**Comment by Kevin222004** [comment]
GitHub, generate report

**Comment by Kevin222004** [comment]
GitHub, generate report

**Review by romani — APPROVED** [review]
Ok to merge

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java [hunk 4]
```diff
@@ -100,6 +102,9 @@
  *
  *      try {} catch (Error e){} finally {} // OK
  *      try {} catch (Error e){} finally{} // violation ''finally' is not followed by whitespace'
+ *
+ *      synchronized(this) { } // violation ''synchronized' is not followed by whitespace'
+ *      synchronized (this) { }
```
Comment (nick-mancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java:107]: Please add `// ok` comment here.

**Review by nick-mancuso — CHANGES_REQUESTED** [review]
Minor item:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java (hunk 4 above)
Comment (Kevin222004) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: done

**Comment by Kevin222004** [comment]
should I have to generate the report again

**Comment by nick-mancuso** [comment]
> should I have to generate the report again ?

No need, documentation update does not require check regression report, only functional updates do.

**Review by nick-mancuso — APPROVED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java (hunk 4 above)
Comment (nick-mancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: Resolved.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterGood.java [hunk 5]
```diff
@@ -36,4 +36,9 @@ public void check5() {
         try {} finally {}
         try {} catch (Exception e){} finally {}
     }
+
+    public void check6() {
+        synchronized (this) { }
+        synchronized (this) { }
```
Comment (strkkk) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterGood.java:42]: what is the point to duplicate line above?

**Review by strkkk — CHANGES_REQUESTED** [review]
please rebase and resolve conflicts

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterGood.java (hunk 5 above)
Comment (Kevin222004) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterGood.java:45]: I have removed it

**Comment by Kevin222004** [comment]
merge conflict resolved
